interfaces/builtin/opencl.go

```diff
@@ -0,0 +1,53 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/ubuntu-core/snappy/interfaces"
+)
+
+const openclConnectedPlugAppArmor = `
+# Description: Can access opencl. 
+# Usage: reserved
+
+  # OpenCL Installable Client Driver (ICD) Loader
+  /etc/OpenCL/vendors/** r,
```

Comment:
Thanks @jdstrand. All, please, don't worry about the text below. 

I am running AMD OpenCL multi-core CPU (auto scale to use all cores). I need try it on 100% clean GPU hardware. It hangs and ```ps u``` STAT says:
S    interruptible sleep (waiting for an event to complete)
l    is multi-threaded (using CLONE_THREAD, like NPTL pthreads do)
\+    is in the foreground process group.

The interface tested
```
  # OpenCL Installable Client Driver (ICD) Loader
  /etc/OpenCL/vendors/ r,
  /etc/OpenCL/vendors/** r,

  # AMD
  /opt/AMDAPP/lib/x86_64/ r,
  /opt/AMDAPP/lib/x86/ r,
  /opt/AMDAPP/lib/x86_64/** r,
  /opt/AMDAPP/lib/x86/** r,
```

dmesg
```
May 23 17:22:32 jucelia-notebook-HP kernel: [32087.007206] audit: type=1326 audit(1464034952.227:65): auid=4294967295 uid=1002 gid=1002 ses=4294967295 pid=9998 comm="john" exe="/snap/john-the-ripper/x1/opencl/john" sig=31 arch=c000003e syscall=203 compat=0 ip=0x7fdd24de1720 code=0x0
May 23 17:23:17 jucelia-notebook-HP kernel: [32132.309280] audit: type=1326 audit(1464034997.532:66): auid=4294967295 uid=1002 gid=1002 ses=4294967295 pid=10013 comm="john" exe="/snap/john-the-ripper/x1/opencl/john" sig=31 arch=c000003e syscall=203 compat=0 ip=0x7f83baa08720 code=0x0
```